### ISSUE TITLE
Ryan's thoughts on Cake

### ISSUE THREAD

**Issue by jrcowan** [devinstormharris/cake#7]
The way I imagine the application working

1. scans database looking for birthdays that are 30 days from current date
2. sends notification asking if you want to track the birthday. 
   -If yes it it adds the person to the track group and gives options for reminders as well a list of hobbies and interests of the user
   -If no it asks if the user would want a notification on the birth date.
        - If yes it adds the person to the OnDay group. 
        - If no it adds the person to the ignore group.
3. Based on the previous question answers it would edit the database entry for the person. 
    - example entry is attached.
4. Scans database and sends out notifications for the appropriate entries. 

I also imagine we would have some kind of webpage that we can send people a link to where they can fill out their info and it will create and entry for them on the database.

![cake](https://user-images.githubusercontent.com/21346588/153473488-a44668dd-0644-4070-9b5f-554fa033231b.PNG)

**Comment by devinstormharris** [comment]
I am considering using the NoSQL database, DynamoDB. This would enable us to have unstructured data storage, so we don't have to intentionally store empty values for "daily count", "3 weeks", etc. "Current age" may not be available as you may only know someone's birth month and day. 

Also, DynamoDB would have key value pairs, and instead of having multiple pairs for {"3 weeks, x"}, {"1 week, x"}, etc, we could just do a configurable value. I.e., { "notification_frequency", "2_weeks" }, or {"notification_frequency", "on_day"}. If we wanted multiple values, we could do { "notification_frequency", "2_weeks,3_weeks,on_day" }.

**Comment by devinstormharris** [comment]
I agree on using a webpage. I wanted to focus on using a console first so we can focus on backend functionality. However, if the backend is designed well, it will not be tightly coupled with the console. This means we can easily add a web frontend without changing the backend much. Perhaps a frontend web application can be our next milestone.
